client/my-sites/subscribers/components/subscribers-header/subscribers-header.tsx

```diff
@@ -58,31 +80,60 @@ export const SubscribersHeader = ( {
 		},
 	};
 
+	const closeSubscriberModal = () => {
+		setShowSubscriberModal( 'none' );
+	};
+
 	return (
-		<NavigationHeader
-			className="stats__section-header modernized-header"
-			title={ translate( 'Subscribers' ) }
-			subtitle={
-				hideSubtitle
-					? null
-					: translate(
-							'Add subscribers to your site and send them a free or {{link}}paid newsletter{{/link}}.',
-							subtitleOptions
-					  )
-			}
-			screenReader={ navItems.insights?.label }
-			navigationItems={ [] }
-		>
-			<Button
-				className="add-subscribers-button"
-				primary
-				disabled={ disableCta }
-				onClick={ () => setShowAddSubscribersModal( true ) }
+		<>
+			<NavigationHeader
+				className="subscribers__header"
+				title={ translate( 'Subscribers' ) }
+				subtitle={
+					hideSubtitle
+						? null
+						: translate(
+								'Add subscribers to your site and send them a free or {{link}}paid newsletter{{/link}}.',
+								subtitleOptions
+						  )
+				}
+				screenReader={ navItems.insights?.label }
+				navigationItems={ [] }
 			>
-				<Gridicon icon="plus" size={ 24 } />
-				<span className="add-subscribers-button-text">{ translate( 'Add subscribers' ) }</span>
-			</Button>
-			<SubscribersHeaderPopover siteId={ selectedSiteId } />
-		</NavigationHeader>
+				<Button
+					variant="primary"
+					className="button add-subscribers-button"
+					disabled={ disableCta }
+					onClick={ () => setShowSubscriberModal( 'add' ) }
+				>
+					<Gridicon icon="plus" size={ 24 } />
+					<span className="add-subscribers-button-text">{ translate( 'Add subscribers' ) }</span>
+				</Button>
+				<SubscribersHeaderPopover
+					siteId={ selectedSiteId }
+					openMigrateSubscribersModal={ () => setShowSubscriberModal( 'migrate' ) }
+				/>
+			</NavigationHeader>
+			{ siteId && (
+				<AddSubscribersModal
+					isVisible={ showSubscriberModal === 'add' }
+					onClose={ closeSubscriberModal }
+					addSubscribersCallback={ ( importError ) => {
+						completeImportSubscribersTask();
+						addSubscribersCallback( importError );
+					} }
+				/>
+			) }
+			{ siteId && (
+				<MigrateSubscribersModal
+					isVisible={ showSubscriberModal === 'migrate' }
+					onClose={ closeSubscriberModal }
+					migrateSubscribersCallback={ ( selectedSourceSiteId, targetSiteId ) => {
+						completeImportSubscribersTask();
+						migrateSubscribersCallback( selectedSourceSiteId, targetSiteId );
+					} }
+				/>
```

Comment:
Added the modals to the header since it seems like the most centrally located component.